cranelift/peepmatic/crates/runtime/src/operator.rs

```diff
@@ -0,0 +1,295 @@
+//! Operator definitions.
+
+use peepmatic_macro::PeepmaticOperator;
+use serde::{Deserialize, Serialize};
+
+/// An operator.
+///
+/// These are a subset of Cranelift IR's operators.
+///
+/// ## Caveats for Branching and Trapping Operators
+///
+/// Branching operators are not fully modeled: we do not represent their label
+/// and jump arguments. It is up to the interpreter doing the instruction
+/// replacement to recognize when we are replacing one branch with another, and
+/// copy over the extra information.
+///
+/// Affected operations: `brz`, `brnz`, `trapz`, `trapnz`.
+#[derive(PeepmaticOperator, Clone, Copy, PartialEq, Eq, Hash, Debug, Serialize, Deserialize)]
+#[repr(u32)]
+pub enum Operator {
+    /// `adjust_sp_down`
+    #[peepmatic(params(iNN), result(void))]
+    AdjustSpDown = 1,
+
+    /// `adjust_sp_down_imm`
+    #[peepmatic(immediates(iNN), result(void))]
+    AdjustSpDownImm,
```

Comment:
Is there any use for peephole optimization of `adjust_sp_down{,_imm}`?

Reply:
Yes, to convert `adjust_sp_down` with a constant argument into an `adjust_sp_down_imm`.